components/FetchedDogs/FetchedDogs.tsx

```diff
@@ -1,119 +1,86 @@
 import React, { useState, useEffect, useRef, useCallback } from "react";
-import { useLocalStorage } from "../../hooks/useLocalStorage";
+import { useFetchDog } from "../../hooks/useFetchDog";
 
-import {
-  DogDetails,
-  DogsDetails,
-  likedDogsDetails,
-  useAppContext,
-} from "../../context";
+import { likedDogsDetails, useAppContext } from "../../context";
 
 import { HeartButton } from "../Buttons/HeartButton/HeartButton";
 import { LoadingSpinner } from "../LoadingSpinner/LoadingSpinner";
 
 import style from "./FetchedDogs.module.css";
 import { Breed } from "../../data/breedsData";
-
-type fetchedDog = {
-  message: string;
-  status: string;
-};
+import Link from "next/link";
 
 export const FetchedDogs = () => {
   const {
+    isMosaic,
     loading,
     setLoading,
     fetchedDogs,
     setFetchedDogs,
     likedDogs,
     setLikedDogs,
     setIsError,
-    apiCallCounter,
-    setApiCallCounter,
+    storagedBreeds,
   } = useAppContext();
 
   const [breed, setBreed] = useState<Breed>();
   const [random, setRandom] = useState(0);
-  const [storagedBreeds] = useLocalStorage<Breed[]>("selectedBreeds", []);
-
+  const [isScreenFilled, setIsScreenFilled] = useState(false);
   const observer = useRef<null | IntersectionObserver>(null);
 
-  const getRandomBreed = () => {
+  const getNextDog = () => {
     setRandom(Math.floor(Math.random() * storagedBreeds.length));
 
     if (random === 0 || random === 1) {
       setRandom(-1);
     }
 
     setBreed(storagedBreeds[random]);
+
     return storagedBreeds[random];
   };
 
+  const waitForAll = () => {
+    const miliSeconds = isMosaic ? 6000 : 2500;
+    setTimeout(() => {
+      setIsScreenFilled(true);
+    }, miliSeconds);
+  };
+  waitForAll();
+
   const useIntersectionObserver = useCallback(
     (node) => {
       if (loading) return;
-      if (observer.current) observer.current.disconnect();
+
+      if (observer.current) {
+        observer.current.disconnect();
+      }
       observer.current = new IntersectionObserver((entries) => {
         if (entries[0].isIntersecting) {
-          const randomBreedFromSelected = getRandomBreed();
-          setBreed(randomBreedFromSelected);
+          getNextDog();
         }
       });
       if (node) observer.current.observe(node);
     },
     [loading]
   );
 
-  const getPosition = (string: string, subString: string, index: number) => {
-    return string.split(subString, index).join(subString).length;
+  const fetchDog = () => {
+    setLoading(true);
+    useFetchDog(storagedBreeds[random]).then((newDog) => {
```

Comment:
What is 'useFetchDog'? If it's a hook, it shouldn't be called here. If it's just a method, it shouldn't be named like a hook.